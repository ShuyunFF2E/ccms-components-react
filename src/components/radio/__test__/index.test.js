import React from 'react';
import renderer from 'react-test-renderer';
import { render, mount } from 'enzyme';
import { prefixCls } from '@utils';
import mountTest from '@tests/shared/mountTest';
import Radio from '../index';

const classSelector = `${prefixCls}-radio`;

describe('Radio', () => {
	mountTest(() => <Radio value={1}>单选</Radio>);

	const Component = (
		<Radio value={1} checked>
			单选
		</Radio>
	);

	it('renders correctly', () => {
		const wrapper = render(Component);
		expect(wrapper).toMatchSnapshot();
	});

	it('mount correctly', () => {
		expect(() => renderer.create(Component)).not.toThrow();
	});

	it('change event correctly', () => {
		const onChange = jest.fn();
		const wrapper = mount(
			<Radio onChange={onChange} value={1}>
				单选
			</Radio>
		);
		wrapper
			.find(`.${classSelector}-input`)
			.at(0)
			.simulate('change');
		expect(onChange).toHaveBeenCalled();
	});

	it('disabled changed invalid', () => {
		const onChange = jest.fn();
		const wrapper = mount(
			<Radio disabled value={1} onChange={onChange}>
				单选
			</Radio>
		);
		wrapper
			.find(`.${classSelector}-input`)
			.at(0)
			.simulate('change');
		expect(onChange).not.toHaveBeenCalled();
	});
});
