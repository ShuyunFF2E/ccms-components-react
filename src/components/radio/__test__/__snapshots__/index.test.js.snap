// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders correctly 1`] = `
<div
  class="cloud-radio-container"
>
  <label
    class="cloud-radio"
  >
    <span
      class="cloud-radio-wrapper"
    >
      <input
        checked=""
        class="cloud-radio-input"
        type="radio"
        value="1"
      />
      <span
        class="cloud-radio-inner"
      />
    </span>
    <span
      class="cloud-radio-text"
    >
      单选
    </span>
  </label>
</div>
`;
